cmd/release/runner.go

```diff
@@ -0,0 +1,168 @@
+	}
+
+	_, err = r.replaceReleaseVersionWithNextWorkInProgress(worktree)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	err = repo.PushContext(ctx, &git.PushOptions{
+		RefSpecs: []config.RefSpec{
+			"refs/heads/*:refs/heads/*",
+			"refs/tags/*:refs/tags/*",
+		},
+	})
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	ghrelease := &github.RepositoryRelease{
+		TagName: github.String(r.flag.TagName),
+	}
+	ghrepositoryRelease, _, err := r.flag.Client.Repositories.CreateRelease(ctx, r.flag.Organization, r.flag.RepositoryName, ghrelease)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	statuses, _, err := r.flag.Client.Repositories.ListStatuses(ctx, r.flag.Organization, r.flag.RepositoryName, commit.String(), &github.ListOptions{})
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	fmt.Printf("Github release '%s' created successfully! %s", r.flag.TagName, *ghrepositoryRelease.HTMLURL)
+	fmt.Println()
+	if len(statuses) > 0 {
+		fmt.Printf("Check that the workflow containing this job ends up successfully %s", *statuses[0].TargetURL)
+		fmt.Println()
+		fmt.Println()
+	}
+
+	return nil
+}
+
+// replaceWorkInProgressVersionWithRelease replaces work in progress version with the release that we want to publish in the source code.
+func (r *runner) replaceWorkInProgressVersionWithRelease(worktree *git.Worktree) (plumbing.Hash, error) {
+	err := replaceVersionInFile(r.flag.WorkInProgressVersion, r.flag.CurrentVersion)
+	if err != nil {
+		return plumbing.Hash{}, microerror.Mask(err)
+	}
+
+	commit, err := addAndCommitChanges(worktree, r.flag.Author, fmt.Sprintf("release %s", r.flag.TagName))
+	if err != nil {
+		return plumbing.Hash{}, microerror.Mask(err)
+	}
+
+	return commit, err
+}
+
+// replaceWorkInProgressVersionWithRelease replaces work in progress version with the release that we want to publish in the source code.
+func (r *runner) replaceReleaseVersionWithNextWorkInProgress(worktree *git.Worktree) (plumbing.Hash, error) {
+	err := replaceVersionInFile(r.flag.CurrentVersion, r.flag.NextPatchWorkInProgressVersion)
+	if err != nil {
+		return plumbing.Hash{}, microerror.Mask(err)
+	}
+
+	commit, err := addAndCommitChanges(worktree, r.flag.Author, fmt.Sprintf("bump new work in progress version to %s", r.flag.NextPatchWorkInProgressVersion))
+	if err != nil {
+		return plumbing.Hash{}, microerror.Mask(err)
+	}
+
+	return commit, err
+}
+
+func addAndCommitChanges(worktree *git.Worktree, author *object.Signature, commitMessage string) (plumbing.Hash, error) {
+	_, err := worktree.Add(VersionFile)
+	if err != nil {
+		return plumbing.Hash{}, microerror.Mask(err)
+	}
+
+	commit, err := worktree.Commit(commitMessage, &git.CommitOptions{
+		Author: author,
+	})
+
+	return commit, microerror.Mask(err)
+}
+
+func replaceVersionInFile(search, replaceWith string) error {
+	sedCommand := fmt.Sprintf("s/%s/%s/", search, replaceWith)
+	command := exec.Command("sed", "-i", "-E", sedCommand, VersionFile)
```

Comment:
I think `-i` here might require some testing / potential workarounds. I have always had problems with it between Mac OS X and rest of the world (mostly GNU sed but not limited to).

See e.g. https://unix.stackexchange.com/a/131940

I'd maybe just implement a function that does that. Just `ReadFile()`, regex replace, `WriteFile()`.

Reply:
fixed PTAL